pkg/runtime/runtime.go

```diff
@@ -1233,13 +1237,30 @@ func (a *DaprRuntime) figureOutComponentCategory(component components_v1alpha1.C
 }
 
 func (a *DaprRuntime) processComponents() {
-	for comp := range a.pendingComponents {
+	for {
+		comp, more := <-a.pendingComponents
+		if !more {
+			a.pendingComponentsDone <- true
+			return
+		}
 		if err := a.processOneComponent(comp); err != nil {
 			log.Errorf("process component %s error, %s", comp.Name, err)
 		}
 	}
 }
 
+func (a *DaprRuntime) flushOutstandingComponents() {
```

Comment:
@wcs1only Can we add simple unit-test for this method?

Reply:
Good call, with proper mocking I should be able to force the race and make sure we actually block until the work is done.